### PR TITLE
Abstract modal into Portal.

### PR DESCRIPTION
**Pull request by fmzbl**

### CODE REVIEW

File: src/shared/portal.rs [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+}
+
+impl PortalView {
+    pub fn show(&mut self, cx: &mut Cx) {
+        self.apply_over(cx, live! {visible: true});
+        self.redraw(cx);
+    }
+
+    pub fn hide(&mut self, cx: &mut Cx) {
+        self.apply_over(cx, live! {visible: false});
```
Comment (jmbejar) [inline src/shared/portal.rs:64]: Nit: there is a `apply_over_and_redraw` https://github.com/makepad/makepad/blob/rik/widgets/src/widget.rs#L542

Comment (fmzbl) [reply src/shared/portal.rs:64]: I tried, but its using this one there: https://github.com/makepad/makepad/blob/rik/platform/src/live_traits.rs#L168

File: src/shared/portal.rs [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+#[derive(Clone, DefaultNone, Eq, PartialEq, Debug)]
+pub enum PortalAction {
+    None,
+    ShowPortalView(LiveId),
+    // ShowPortalViewWithProps(Box<PortalActionWithProps>),
+    Close,
+}
+
+// #[derive(Debug)]
```
Comment (jmbejar) [inline src/shared/portal.rs:103]: Commented put code to remove

File: src/shared/portal.rs [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+        }
+
+        portal_view_ref.show(cx);
+        self.active_portal_view = ActivePortalView::Active(portal_view_id);
+
+        self.redraw(cx);
+        Ok(())
+    }
+
+    // pub fn show_portal_view_by_id_with_props<T>(
```
Comment (jmbejar) [inline src/shared/portal.rs:265]: Commented out code to remove

**Review by jmbejar — APPROVED** [review]
I'd remove the commented out code before merging. However, LGTM 👍

**Pull request merged**